thirdparty/misc/easing_equations.cpp

```diff
@@ -1,323 +0,0 @@
-
-static real_t in(real_t t, real_t b, real_t c, real_t d) {
-	return c - out(d - t, 0, c, d) + b;
-}
-
-static real_t out(real_t t, real_t b, real_t c, real_t d) {
-	if ((t /= d) < (1 / 2.75f)) {
-		return c * (7.5625f * t * t) + b;
-	} else if (t < (2 / 2.75f)) {
-		float postFix = t -= (1.5f / 2.75f);
-		return c * (7.5625f * (postFix)*t + .75f) + b;
-	} else if (t < (2.5 / 2.75)) {
-		float postFix = t -= (2.25f / 2.75f);
-		return c * (7.5625f * (postFix)*t + .9375f) + b;
-	} else {
-		float postFix = t -= (2.625f / 2.75f);
-		return c * (7.5625f * (postFix)*t + .984375f) + b;
-	}
-}
-
-static real_t in_out(real_t t, real_t b, real_t c, real_t d) {
-	return (t < d / 2) ? in(t * 2, b, c / 2, d) : out((t * 2) - d, b + c / 2, c / 2, d);
-}
-
-static real_t out_in(real_t t, real_t b, real_t c, real_t d) {
-	return (t < d / 2) ? out(t * 2, b, c / 2, d) : in((t * 2) - d, b + c / 2, c / 2, d);
-}
-}; // namespace bounce
-///////////////////////////////////////////////////////////////////////////
-// back
-///////////////////////////////////////////////////////////////////////////
-namespace back {
-static real_t in(real_t t, real_t b, real_t c, real_t d) {
-	float s = 1.70158f;
-	float postFix = t /= d;
-	return c * (postFix)*t * ((s + 1) * t - s) + b;
-}
-
-static real_t out(real_t t, real_t b, real_t c, real_t d) {
-	float s = 1.70158f;
-	t = t / d - 1;
-	return c * (t * t * ((s + 1) * t + s) + 1) + b;
-}
-
-static real_t in_out(real_t t, real_t b, real_t c, real_t d) {
-	float s = 1.70158f * 1.525f;
-	t /= d / 2;
-	if (t < 1) return c / 2 * (t * t * ((s + 1) * t - s)) + b;
-	t -= 2;
-	return c / 2 * (t * t * ((s + 1) * t + s) + 2) + b;
-}
-
-static real_t out_in(real_t t, real_t b, real_t c, real_t d) {
-	return (t < d / 2) ? out(t * 2, b, c / 2, d) : in((t * 2) - d, b + c / 2, c / 2, d);
-}
-}; // namespace back
-
-Tween::interpolater Tween::interpolaters[Tween::TRANS_COUNT][Tween::EASE_COUNT] = {
-	{ &linear::in, &linear::out, &linear::in_out, &linear::out_in },
-	{ &sine::in, &sine::out, &sine::in_out, &sine::out_in },
-	{ &quint::in, &quint::out, &quint::in_out, &quint::out_in },
-	{ &quart::in, &quart::out, &quart::in_out, &quart::out_in },
-	{ &quad::in, &quad::out, &quad::in_out, &quad::out_in },
-	{ &expo::in, &expo::out, &expo::in_out, &expo::out_in },
-	{ &elastic::in, &elastic::out, &elastic::in_out, &elastic::out_in },
-	{ &cubic::in, &cubic::out, &cubic::in_out, &cubic::out_in },
-	{ &circ::in, &circ::out, &circ::in_out, &circ::out_in },
-	{ &bounce::in, &bounce::out, &bounce::in_out, &bounce::out_in },
-	{ &back::in, &back::out, &back::in_out, &back::out_in },
-};
-
-real_t Tween::_run_equation(TransitionType p_trans_type, EaseType p_ease_type, real_t t, real_t b, real_t c, real_t d) {
-	if (d == 0) {
-		// Special case to avoid dividing by 0 in equations.
-		return b + c;
-	}
-
-	interpolater cb = interpolaters[p_trans_type][p_ease_type];
-	ERR_FAIL_COND_V(cb == NULL, b);
```

Comment:
This null check is still needed, because `Tween.interpolate_property()` doesn't check validity of the provided ease/transition. SceneTreeTween doesn't have this problem.